pkgs/moduleit/module-definition.nix

```diff
@@ -4,6 +4,33 @@ with lib;
 
 let
 
+  initializerModule = { name, config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The name of the initializer. An initializer is usually used to scaffold
+          a project by creating some files and/or directories.
+        '';
+      };
+
+      start = mkOption {
+        type = commandType;
+        description = lib.mdDoc ''
+          The command to run the initializer.
+        '';
+      };
+
+      runOnce = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          If runOnce is true, the initializer will only be run once.
+        '';
```

Comment:
how do you intend for users to use initializers? I'm not sure what this is for, or if it should be included. eg for NPM, `npm init` may be run for every package in a workspace, as with `cargo init`. I'm not sure there are many initializers that break when the initializer is run multiple times in a directory ie subdirectories

Reply:
The initializer is run once when the module activates, even if it's re-activated. There will be a cache file to remember whether it ran or not. If runOnce is false, then it will get run every time the module activates, so it's on the initializer to remember whether it ran or not.